components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -161,15 +158,85 @@ class Nimbus : NimbusApi {
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        // We need to launch this off of the main thread because it involves a network request
+        // within the nimbus-sdk.
+        scope.launch {
+            if (tryUpdateExperiments()) {
+                tryGetActiveExperiments()?.let {
+                    recordExperimentTelemetry(it)
+                    onExperimentUpdated?.invoke(it)
+                }
+            }
+        }
     }
 
     override fun optOut(experimentId: String) {
         if (!isInitialized) return
         nimbus.optOut(experimentId)
     }
 
+    // Wraps the `Nimbus.updateExperiments()` in a try/catch since the underlying nimbus-sdk
+    // makes a network request that might throw an error.  This also wraps the call in
+    // a simple retry mechanism before giving up.
+    // Returns true if the update completed without errors, false otherwise.
+    // Note: a return of "true" doesn't mean that there was any new experiments, just that there
+    // were no errors.
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun tryUpdateExperiments(): Boolean {
+        var retries = 0
+        while (retries++ < MAX_RETRY_COUNT) {
+            // Get experiments
+            try {
+                nimbus.updateExperiments()
+                return true
+            } catch (e: ErrorException.RequestError) {
+                Log.log(Log.Priority.ERROR,
+                    LOG_TAG,
+                    message = "Error fetching experiments from endpoint: $e"
+                )
+            } catch (e: ErrorException.InvalidExperimentResponse) {
+                Log.log(Log.Priority.ERROR,
+                    LOG_TAG,
+                    message = "Invalid experiment response: $e"
+                )
+            }
+
+            Thread.sleep(RETRY_DELAY_MS)
```

Comment:
nit: Since you are calling this from a coroutine scope, you could consider making this a `suspend` function and use `delay()` here. Which would have the advantage to not block the thread. That may be helpful if you intent to do other things on this scope while this is waiting. Otherwise the thread would be blocked. I do not know about the internals here, so I do not know if that would make sense though. :)

Reply:
After some discussion, I am going to remove the retries from this PR.  It's not really clear that the Nimbus component should be responsible for the retry behavior and should probably just surface an error so that the application can decide how to proceed.